mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -380,7 +380,7 @@ def SYCLGetIDResult : AnyTypeOf<[I64, SYCL_IDType]>;
 def SYCLGetRangeResult : AnyTypeOf<[I64, SYCL_RangeType]>;
 def SYCLIndexSpaceGetIDResult : AnyTypeOf<[Index, SYCL_IDType]>;
 def SYCLIndexSpaceGetRangeResult : AnyTypeOf<[Index, SYCL_RangeType]>;
-def SYCLGetResult : AnyTypeOf<[I64, MemRefOf<[I64]>]>;
+def SYCLGetResult : AnyTypeOf<[I64, MemRefOf<[I64]>, I32, MemRefOf<[I32]>]>;
```

Comment:
This should be `AnyTypeOf<[Index, MemRefOf<[Index]>]>` IMO, but that would require work in codegen. Can you put a comment here?